### PR TITLE
Add XClass

### PR DESCRIPTION
**Pull request by Snazzah**
### Information

> Plugin Name:  XClass<br>
> DI Version Tested On:  5.2.0<br>

### Systems Tested On

Client:
 - [ ] Stable
 - [ ] PTB
 - [x] Canary
 - [x] Development

OS:
 - [x] Windows
 - [x] Mac
 - [x] Linux (include distro)

### Plugin Description

Adds classes to elements that can define more about that element.

### CODE REVIEW

File: XClass/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        this.log('Loading listeners');
+        this.userModalBind = this.userModal.bind(this);
+        this.userPopoutBind = this.userPopout.bind(this);
+        this.messageGroupBind = this.messageGroup.bind(this);
+        window.A.Watcher.on('messageGroup', this.messageGroupBind);
+        window.A.Watcher.on('userPopout', this.userPopoutBind);
+        window.A.Watcher.on('userModal', this.userModalBind);
+    }
+
+    static get after() { return ['AFunc'] }
```
Comment (cking) [inline XClass/index.js]: all lower case. it's using the internal plugin manager representation and that ones using all lower case versions

File: XClass/config.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "color": "8e0cd3"
```
Comment (cking) [inline XClass/config.json:2]: Try to not include the config.json

File: XClass/index.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+const Plugin = module.parent.require('../Structures/Plugin');
+
+class XClass extends Plugin {
+    constructor(...args) {
```
Comment (cking) [inline XClass/index.js:4]: Instead of `constructor(...args) ` use `load()`

**Review by cking — CHANGES_REQUESTED** [review]

File: XClass/index.js (hunk 3 above)
Comment (Ratismal) [reply XClass/index.js:4]: using load makes more semantic sense but either are fine. it should be noted that load will perform faster than the constructor as it doesn't have to use the rest operator

**Pull request merged**